react/components/FilterOptionTemplate.js

```diff
@@ -29,16 +30,46 @@ const FilterOptionTemplate = ({
   children,
   filters,
   initiallyCollapsed = false,
+  truncateFilters = false,
 }) => {
   const [open, setOpen] = useState(!initiallyCollapsed)
   const handles = useCssHandles(CSS_HANDLES)
+  const [truncated, setTruncated] = useState(true)
+  const maxItemsThreshold = 12
 
   const renderChildren = () => {
     if (typeof children !== 'function') {
       return children
     }
-
-    return filters.map(children)
+    return (
+      <>
+        {filters
+          .slice(
+            0,
+            truncateFilters && filters.length >= maxItemsThreshold && truncated
```

Comment:
`truncateFilters && filters.length >= maxItemsThreshold`

Maybe this can be a variable, it's used twice, it will improve the legibility.